### PR TITLE
Provide Container Registry scraper

### LINKED ISSUE #325: Provide support for scraping metrics for Azure Container Registry
Provide support for scraping metrics for Azure Container Registry

### PR DESCRIPTION
**Pull request by tomkerkhove**
Provide Container Registry scraper

Closes #325

### Configuration
```yaml
azureMetadata:
  tenantId: c8819874-9e56-4e3f-b1a8-1c0325138f27
  subscriptionId: 0f9d7fea-99e8-4768-8672-06a28514f77e
  resourceGroupName: promitor-sources
metricDefaults:
  aggregation:
    interval: 01:00:00
metrics:
  - name: demo_generic_total_pull_count
    description: "Amount of images that were pulled from the container registry"
    resourceType: ContainerRegistry
    registryName: promitor
    azureMetricConfiguration:
      metricName: TotalPullCount
      aggregation:
        type: Average
```

### Output
```yaml
# HELP demo_generic_total_pull_count Amount of images that were pulled from the container registry
# TYPE demo_generic_total_pull_count gauge
demo_generic_total_pull_count 1 1553635305491
```
### Portal
image

### CODE REVIEW

File: src/Promitor.Core.Scraping/Configuration/Model/ResourceType.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@ public enum ResourceType
         Generic = 2,
         StorageQueue = 3,
         ContainerInstance = 4,
-        VirtualMachine = 5
+        VirtualMachine = 5,
+        ContainerRegistry = 6
```
Comment (brandonh-msft) [inline src/Promitor.Core.Scraping/Configuration/Model/ResourceType.cs]: I'm beginning to wonder if having the explicit values here will hamper merges in the future... thoughts?

Comment (tomkerkhove) [reply src/Promitor.Core.Scraping/Configuration/Model/ResourceType.cs]: Good point, maybe better to remove it?

Comment (tomkerkhove) [reply src/Promitor.Core.Scraping/Configuration/Model/ResourceType.cs]: Actually I would suggest keeping them in case we want to re-order it and be safe

**Pull request merged**